lib/manager/mix/artifacts.spec.ts

```diff
@@ -96,8 +97,25 @@ describe(getName(), () => {
     expect(execSnapshots).toMatchSnapshot();
   });
 
+  it('returns updated mix.lock in subdir', async () => {
+    setAdminConfig({ ...adminConfig, binarySource: 'docker' });
+    fs.writeLocalFile.mockResolvedValueOnce('New mix.exs');
+    fs.getSiblingFileName.mockReturnValueOnce('subdir/mix.lock');
+    const execSnapshots = mockExecAll(exec);
+    expect(
+      await updateArtifacts({
+        packageFileName: 'subdir/mix.exs',
+        updatedDeps: [{ depName: 'plug' }],
+        newPackageFileContent: '{}',
+        config,
+      })
+    ).toMatchSnapshot();
```

Comment:
Do not use snapshot test, when you expect null. Use `toBeNull()` matcher.

Shouldn't we expected some non null response here?

Reply:
I'm not sure it much matters? I'm basically just testing whether it hits the `expect(fs.readLocalFile).toBeCalledWith('subdir/mix.lock', 'utf8');` line with the correct subdirectory. If that's correct, I don't think we much care about the rest of the update method, since that's already tested in the various other methods here and doesn't change behaviour for the subdir usecase.

Reply:
Ok,then use `toBeNull` matcher if the result is null anyways.